src/ops.rs

```diff
@@ -588,9 +588,47 @@ fn op_open(
   let cmd_id = base.cmd_id();
   let inner = base.inner_as_open().unwrap();
   let filename = PathBuf::from(inner.filename().unwrap());
-  // TODO let perm = inner.perm();
+  //  let perm = inner.perm();
+  let mode = inner.mode().unwrap();
 
-  let op = tokio::fs::File::open(filename)
+  let mut open_options = tokio::fs::OpenOptions::new();
+
+  match mode {
+    "r" => {
+      open_options.read(true);
+    }
+    "r+" => {
+      open_options.read(true).write(true);
+    }
+    "w" => {
+      open_options.create(true).write(true).truncate(true);
+    }
+    "w+" => {
+      open_options
+        .read(true)
+        .create(true)
+        .write(true)
+        .truncate(true);
+    }
+    "a" => {
+      open_options.create(true).append(true);
+    }
+    "a+" => {
+      open_options.read(true).create(true).append(true);
+    }
+    "x" => {
+      open_options.create_new(true).write(true);
+    }
+    "x+" => {
+      open_options.create_new(true).read(true).write(true);
+    }
+    &_ => {
+      panic!("Unknown file open mode.");
+    }
+  }
+
```

Comment:
Should I move this whole operation somewhere else?

Probably some comments explaining why each option is used would be appropriate as well.